python/PyQt6/core/__init__.py.in

```diff
@@ -598,6 +625,33 @@ except ModuleNotFoundError:
        raise QgsNotSupportedException('QgsRasterLayer.as_numpy is not available, numpy is not installed on the system')
 
    QgsRasterLayer.as_numpy = _raster_layer_as_numpy
+   
+   def _geometry_as_numpy(self):
+      raise QgsNotSupportedException('QgsGeometry.as_numpy is not available, numpy is not installed on the system')
+
+   QgsGeometry.as_numpy = _geometry_as_numpy
+
+try:
+   import shapely as _shapely
+
+   def _geometry_as_shapely(self) -> _shapely.geometry.base.BaseGeometry:
+      wkb_qbytearray = self.asWkb()  # Get the geometry in WKB format (QByteArray)
+      wkb_bytes = bytes(wkb_qbytearray)
+      shapely_geom = _shapely.from_wkb(wkb_bytes)
```

Comment:
Does this work? It would save a copy of the data, and might squeeze out a bit more performance:

```suggestion
      shapely_geom = _shapely.from_wkb(wkb_qbytearray.data())
```